Fixing Up Things

Pop000100 opened the pull request:
Fixing up things I have missed before 1.4.4
- Start transitioning to NextBool instead Next since it is 1 character shorter, and I think more readable.
- Missed using TileEntity entitysource on the autofisher and chimney(This can improve mod compatibility, no other effect).
- 1.4.0.1 made hellforges only mineable with 60+ pick power in the underworld. So I copied changes for the ancient one.
- Change the glowmask system to not use the vanilla array. This has been causing problems. I implemented it like the TModloader team recommends improving mod compatibility.
- Fix issues of the Red Chimney caused issues saving, the velocity thing you had seems to be overwritten by the goreloader so I removed it.

PS. ChimneySmokes have special AI that runs that includes assignment making any changes made here useless. Must have been a 1.4 addition.

Pop000100 committed:
Fixing Up Things

Fixing up things I have missed before 1.4.4
- Start transitioning to NextBool instead Next since it is 1 character shorter, and I think more readable
- Missed using TileEntity entitysource on the autofisher and chimney(this can improve mod compatibility)
- 1.4.3 made hellforges only mineable with 60+ pick power in the underworld. So I copied changes for the ancient one.
- Change the glowmask system to not use the vanilla array. this has been causing problems. I implemented it like the TModloader team recommends improving mod compatibility.
- Fix issues of the Red Chimney caused issues saving, the velocity thing you had seems to be overwriting by the goreloader so I removed it.

PS. ChimneySmokes have special AI that runs that includes assignment making any changes made here useless. Must have been a 1.4 addition.

gardenappl commented on Items/Weapons/UndyingSpear.cs:
```diff
@@ -40,9 +42,28 @@ public override void SetDefaults()
 		}
-		
+
+		public override void PostDrawInInventory(SpriteBatch spriteBatch, Vector2 position, Rectangle frame, Color drawColor, Color itemColor, Vector2 origin, float scale)
```
If you really have to use custom drawing code for every glowmask item, then it's probably better to move all the repeated code elsewhere. Maybe a subclass of ModItem like GlowmaskModItem that has its own Texture2D modGlowMask field.

Pop000100 replied:
Done. I think I found a better name for it. Also the issue was that the old method conflicted with another mod was doing the same thing so it expected the size to be the same since the load. The other mod changed the size causing out of bounds exception on unload. I probably could have fixed it up but this will never have conflicts with reasonable code.

gardenappl commented:
I'm sorry for the delay. This looks good, I did leave one comment though.

> PS. ChimneySmokes have special AI that runs that includes assignment making any changes made here useless. Must have been a 1.4 addition.

Does this mean we can remove some of our code and it will work? That'd be nice. I'm pretty sure the red chimney's code was just copy-pasted from decompiled Terraria and I'd rather have less of that kind of thing.

Pop000100 commented on Tiles/RedChimneyTE.cs:
```diff
@@ -27,26 +28,23 @@ public enum State : byte
 				{
-					smokeVel.X = -Main.windSpeedCurrent;
+					Gore.NewGore(new EntitySource_TileEntity(this), smokePos, smokeVel, smokeType, Main.rand.NextFloat() * 0.4f + 0.4f);
 				}
```
The if state and the change in velocity are what became redundant, cause their behavior is hardcoded that is why I removed them.


Edit: That is weird selection of code but alright.

Pop000100 committed:
Abstract Items with GlowMasks

- Create base class for Items with glowmasks.
- Use said base class.

PS. Projectiles are quite different and there is one that would use it. so no class for them.

gardenappl commented on MiscGlowMasks.cs:
```diff
@@ -9,53 +10,64 @@ namespace GoldensMisc
+
+	public abstract class GlowingModItem : ModItem
+    {
+		private Texture2D _glowMask;
```
My C# skills are getting rusty and I can't tell why a public setter and getter is useful.

Pop000100 replied:
Technically it is more compatible for some types of binaries but I can just make glowMask public, I have been working on various projects on bepinex where it is more common since those are designed as librarys. I just wrote that way, It isn't necessary and it is not stylistic for this project.

Ill switch to just a public field.

gardenappl replied:
I was more so hinting that maybe the setter could be private. But this works too.

Pop000100 committed:
Change to just a public field

- To match the rest of the code.

Pop000100 commented:
I asked this in our conversation on 1.4.4 on the Terraria forums

What do you think about disabling the redundant things by default and leaving them as is? Cause the way I see it is having more options is better, some people might like them even if there is a item that is obtained in a different place with mostly the same use.

I will add that I don't mind the upkeep cost.

Also I hope you are doing well, I know you said you weren't close to the conflict zones last time I asked you.

gardenappl commented:
I left a response in the forum DMs. I'm okay, just got distracted by some things, sorry. PR looks good, merging.
